Python/ceval.c

```diff
@@ -2124,6 +2124,17 @@ _PyEval_UnpackIterableStackRef(PyThreadState *tstate, _PyStackRef v_stackref,
         _PyErr_Format(tstate, PyExc_ValueError,
                       "too many values to unpack (expected %d)",
                       argcnt);
+        PyObject *exc = _PyErr_GetRaisedException(tstate);
+        _Py_DECLARE_STR(_unpacked_value, "_unpacked_value")
+        _Py_DECLARE_STR(_unpack_expected_argcnt, "_unpack_expected_argcnt")
+        if (PyObject_SetAttr(exc, &_Py_STR(_unpacked_value), v))
+            goto Error;
+        if (PyObject_SetAttr(exc,
+                             &_Py_STR(_unpack_expected_argcnt),
+                             PyLong_FromLong(argcnt)))
+            goto Error;
+
+        _PyErr_SetRaisedException(tstate, exc);
```

Comment:
- Oops, I mixed up `Py_ID` and `Py_STR` in the suggestion. I should have caught that before commenting, sorry!
  `Py_ID` doesn't need the DECLARE.
- `exc = _PyErr_GetRaisedException(...)` gives you ownership of the `exc` reference; you need to re-raise it, chain it or at least DECREF it.
- `if` bodies need to be surrounded by braces, per PEP-7.
```suggestion
        PyObject *exc = _PyErr_GetRaisedException(tstate);
        if (PyObject_SetAttr(exc, &_Py_ID(_unpacked_value), v)) {
            _PyErr_ChainExceptions1(exc);
            goto Error;
        }
        if (PyObject_SetAttr(exc,
                             &_Py_ID(_unpack_expected_argcnt),
                             PyLong_FromLong(argcnt))) {
            _PyErr_ChainExceptions1(exc);
            goto Error;
        }

        _PyErr_SetRaisedException(tstate, exc);
```
(After switching to `_Py_ID` you'll need to re-run `make regen-global-objects`.)

Reply:
Thanks for all the help!